dom.bs

```diff
@@ -1393,6 +1393,9 @@ property of the event being dispatched.
      <li><p>Otherwise, set <var>event</var>'s {{Event/eventPhase}} attribute to
      {{Event/CAPTURING_PHASE}}.
 
+     <li><p>Run {light dismiss open pop-ups} given <var>struct</var>'s <a
+     for=Event/path>invocation target</a>.
```

Comment:
This needs some kind of note explaining the setup. This also seems like a layering violation. Is this really supposed to happen for all events or just certain UI events? Is it supposed to happen for synthetic events or just events dispatched by the user agent?

Reply:
> This also seems like a layering violation.

Is HTML supposed to refer to DOM but not the other way around? I noticed that this spec refers to HTMLSlotElement from the HTML spec...

> Is this really supposed to happen for all events or just certain UI events?

Yes, it really happens for all events, at least in the [chromium implementation](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/dom/node.cc;l=2885;drc=b5666c58a9cf7539e6f57c3e6e957c8fd956c650).

> (If you rebase and force push the CI error will go away.)

done

Reply:
It seems to me that the implementation only does something for mouse and keyboard input. Although maybe also for synthetic mouse and keyboard input? It still strikes me as somewhat wrong to have it as part of event dispatch. What's the design motivation for that?

Reply:
There has been some discussion about the constraints of when to actually light dismiss based on the event here, hopefully the new changes address this concern: https://github.com/whatwg/html/issues/7785#issuecomment-1299086185

> It seems to me that the implementation only does something for mouse and keyboard input.

Do you mean that it should only work if the event is trusted? The implementation does not currently check isTrusted, and [mason's new patch](https://chromium-review.googlesource.com/c/chromium/src/+/3995612/1/third_party/blink/renderer/core/html/html_element.cc#1744) doesn't do so either.

> Although maybe also for synthetic mouse and keyboard input? It still strikes me as somewhat wrong to have it as part of event dispatch. What's the design motivation for that?

I don't know what the design motivation was, but if you have any alternative ideas I'm all ears